src/shared/reportIssue.ts

```diff
@@ -0,0 +1,142 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Extension, vscode } from "../vscodeAdapter";
+import { Options } from "../shared/options";
+import { IHostExecutableResolver } from "../shared/constants/IHostExecutableResolver";
+import { basename, dirname } from "path";
+import { DotnetInfo } from "./utils/getDotnetInfo";
+
+const issuesUrl = "https://github.com/dotnet/vscode-csharp/issues/new";
+
+export default async function reportIssue(
+    vscode: vscode,
+    csharpExtVersion: string,
+    getDotnetInfo: (dotNetCliPaths: string[]) => Promise<DotnetInfo>,
+    shouldIncludeMonoInfo: boolean,
+    options: Options,
+    dotnetResolver: IHostExecutableResolver,
+    monoResolver?: IHostExecutableResolver) {
+    // Get info for the dotnet that the language server executable is run on, not the dotnet the language server will execute user code on.
+    let fullDotnetInfo: string | undefined;
+    try {
+        const info = await dotnetResolver.getHostExecutableInfo(options);
+        const dotnetInfo = await getDotnetInfo([ dirname(info.path) ]);
+        fullDotnetInfo = dotnetInfo.FullInfo;
+    } catch (error) {
+        const message = error instanceof Error ? error.message : `${error}`;
+        fullDotnetInfo = message;
+    }
+
+    let monoInfo = "";
+    if (shouldIncludeMonoInfo && monoResolver) {
+        monoInfo = await getMonoIfPlatformValid(options, monoResolver);
+    }
+    
+    let extensions = getInstalledExtensions(vscode);
+
+    const useOmnisharp = options.commonOptions.useOmnisharpServer;
+    const logInfo = getLogInfo(useOmnisharp);
+
+    const body = `## Issue Description ##
+## Steps to Reproduce ##
+
+## Expected Behavior ##
+
+## Actual Behavior ##
+
+## Logs ##
+
+${logInfo}
+
+## Environment information ##
+
+**VSCode version**: ${vscode.version}
+**C# Extension**: ${csharpExtVersion}
+**Using OmniSharp**: ${useOmnisharp}
+
+${monoInfo}
+<details><summary>Dotnet Information</summary>
+${fullDotnetInfo}</details>
+<details><summary>Visual Studio Code Extensions</summary>
+${generateExtensionTable(extensions)}
+</details>
+`;
+
+    const queryStringPrefix: string = "?";
+    const issueDefault = "Please paste the output from your clipboard";
```

Comment:
Do I dare ask why we don't just fill in the body directly since we have a way to do that?

Reply:
I think it is because of potential URI escaping issues (the body gets put into the actual URI).  Might be possible still, I can investigate it later.  For now I'll leave it since this is the existing behavior.

Reply:
Yeah, I don't know why we can't just call an escape method here. :-/

Reply:
its not as simple as doing encodeURIComponent and passing it along, I did try that, but you get this:
<img width="419" alt="image" src="https://github.com/dotnet/vscode-csharp/assets/5749229/03d814ad-21d2-44b9-8599-b153ec875c4e">

So will dig into this more later.

Reply:
Great suggestion.  I'm using that
<img width="937" alt="image" src="https://github.com/dotnet/vscode-csharp/assets/5749229/fd38d551-11b7-4cc2-9582-4a9f4e1263e9">